### PR TITLE
week 1 assessment

### PR DESCRIPTION
**Pull request by thomasgonda3**
I'm not 100% sure the testing with jest file is there, but I think it is.  Otherwise, I think it's done.

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -10,23 +10,48 @@ var temp1 = 35
...
 // Use the test variable provided below. Expected outcome: [15, 35, 0, 30, -45]
 
 var myNumbers1 = [3, 7, 0, 6, -9]
 
+function iteratorMultiplier (array) {
+  for (let value in array)  {
+    array[value] *= 5;
+  }
+  return array;
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:40]: this is a super interesting solution. however, be aware that what you're doing is reassigning the values in the original array. there are going to be situations where you'll want to return _new_ values while keeping the original data intact.

File: code-challenges.js [hunk 2]
```diff
@@ -10,23 +10,48 @@ var temp1 = 35
...
 
 
 // --------------------3) Create a function that multiplies each number in the array by 5 using map.
 // Use the test variable provided below. Expected outcome: [40, -35, 0, 30, 10]
 
 var myNumbers2 = [8, -7, 0, 6, 2]
 
+function mapMultiplier (array)  {
+  return array.map(x => x * 5);
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:52]: great!

File: code-challenges.js [hunk 3]
```diff
@@ -35,8 +60,15 @@ var myNumbers2 = [8, -7, 0, 6, 2]
 var stringWithVowels1 = "HeyThereLearnStudent"
 var stringWithVowels2 = "ILoveJavaScript"
 
+function vowelRemover (string)  {
+  //an array of vowels for the filter test to use.
+  let vowelTest = ['a','e','i','o','u'];
+  //if the string's current character does not exist within the array vowels, it passes the filter test.
+  return string.split('').filter(value => vowelTest.indexOf(value.toLowerCase()) == -1).join('');
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:68]: nice pseudocode on a single-line solution!

File: code-challenges.js [hunk 4]
```diff
@@ -46,6 +78,16 @@ var notAString1 = true
 var notAString2 = 42
 
 
+function vowelRemover2 (string)  {
+  if (typeof(string) != "string") return "Not a string."
```
Comment (rachaelstanislaw) [inline code-challenges.js:82]: yay! you can even save yourself a couple key strokes if you'd like, you can omit the parentheses after `typeof`

File: code-challenges.js [hunk 5]
```diff
@@ -54,7 +96,27 @@ var notAString2 = 42
...
 var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
 
+function onlyCats(array)  {
+  let answer = [];
+  //if the animal is a cat, move it to the array we will return
+  for (let obj of array)  {
+    if (obj.animal == 'cat') answer.push(obj);
+  }
+  return answer;
+}
```
Comment (rachaelstanislaw) [inline code-challenges.js:106]: nice job!

File: code-challenges.js [hunk 6]
```diff
@@ -54,7 +96,27 @@ var notAString2 = 42
...
+
+function nonCatNames (array)  {
+  let string = "";
+  //add the names of animals that are not cats onto the end of a string that we will return.
+  for (let obj of array)  {
+    if (obj.animal != 'cat') string = string.concat(`"${obj.name}" `);
+  }
+  return string;
+}
+
```
Comment (rachaelstanislaw) [inline code-challenges.js:121]: nice work! i would only challenge you to refactor this using higher order functions

File: interview-questions.md [hunk 7]
```diff
@@ -4,64 +4,68 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. What is a function? Why would you use one?
 
-  Your answer:
+  Your answer: A reusable piece of code that takes in different arguments depending on the situation.
```
Comment (rachaelstanislaw) [inline interview-questions.md:7]: ✔️

File: interview-questions.md [hunk 8]
```diff
@@ -4,64 +4,68 @@ Answer the following questions. First, without external resources. Challenge you
...
 
-  Researched answer:
+  Researched answer:  A function is a JavaScript procedure—a set of statements that performs a task or calculates a value.
 
 
 
 2. What is the difference between map and filter?
 
-  Your answer:
+  Your answer: Map forces every element in the array to undergo a function, then returns the new array.  Filter returns an array that removes every element in the array that does not pass a specified test.
```
Comment (rachaelstanislaw) [inline interview-questions.md:15]: ✔️

File: interview-questions.md [hunk 9]
```diff
@@ -4,64 +4,68 @@ Answer the following questions. First, without external resources. Challenge you
...
 
-  Researched answer:
+  Researched answer: filter() returns a subset of the elements from the original array, while . map() produces an array with new, different entries based on the elements in the original array.
 
 
 
 3. What is the difference between console.log() and return?
 
-  Your answer:
+  Your answer: console.log outputs the evaluation of whatever argument is within itself, and does not actually change the state of the program.  When you return, you exit a function and give back an answer.
```
Comment (rachaelstanislaw) [inline interview-questions.md:23]: ✔️

File: interview-questions.md [hunk 10]
```diff
@@ -4,64 +4,68 @@ Answer the following questions. First, without external resources. Challenge you
...
 
-  Researched answer:
+  Researched answer:  console.log logs the arguments passed to the web console, return will exit a function and also call back the value to whatever invoked the function.
 
 
 
 4. In regards to functions, what is an argument?
 
-  Your answer:
+  Your answer: The variable input that is given when a function is run that makes it reusable.
```
Comment (rachaelstanislaw) [inline interview-questions.md:31]: ✔️

File: interview-questions.md [hunk 11]
```diff
@@ -4,64 +4,68 @@ Answer the following questions. First, without external resources. Challenge you
...
 
-  Researched answer:
+  Researched answer:  A value that is passed between programs, subroutines or functions.  They are independent items, or variables that contain data or codes.
 
 
 
 5. Give a brief description of proper pair programming techniques. What are the roles of each person?
 
-  Your answer:
+  Your answer:  The driver types all of the code and makes quick edits if he sees a problem.  The passenger directs where the code is headed and also points out any errors in the code he sees.
```
Comment (rachaelstanislaw) [inline interview-questions.md:39]: ✔️ or she! ;)

File: interview-questions.md [hunk 12]
```diff
@@ -4,64 +4,68 @@ Answer the following questions. First, without external resources. Challenge you
...
 
 
 
 6. What is TDD? Describe the work flow associated with TDD.
 
-  Your answer:
+  Your answer:  It's when you write tests for your code so that you know it works within several different scenarios.
 
-  Researched answer:
+  Researched answer:  The work flow is write the test, fail the test, write codes, test passes, then refactor your code.
```
Comment (rachaelstanislaw) [inline interview-questions.md:49]: ✔️

File: interview-questions.md [hunk 13]
```diff
@@ -4,64 +4,68 @@ Answer the following questions. First, without external resources. Challenge you
...
 
 
 
 7. What is something we did in class this week you found helpful?  
 
-  Your answer:
+  Your answer:  I thought the pair programming was really fun, I didn't know how it would feel before doing it but I really enjoyed myself when going through.  I feel like when we were going through lectures and I wanted to interject sometimes everyone would just talk so quickly I didn't really have a chance but when I posted in the chat my question was usually answered so that was pretty useful.
```
Comment (rachaelstanislaw) [inline interview-questions.md:55]: ✔️ great feedback! we'll note it and try and slow down a little to allow for more conversation during the lectures themselves

File: interview-questions.md [hunk 14]
```diff
@@ -4,64 +4,68 @@ Answer the following questions. First, without external resources. Challenge you
...
+
+- Yarn - Yarn is new and Open Source JavaScript package manager developed by Facebook. Yarn is fully compatible with the npm registry and can work alongside npm, but it's aim it to be a safer, more secure and more reliable alternative.
+
+- React State - React components has a built-in state object.  The state object is where you store property values that belongs to the component.  When the state object changes, the component re-renders.
+
+- CRUD - CRUD stands for Create, Read, Update and Delete and these are four basic functions to manipulate data in a database.  The CRUD paradigm is common in constructing web applications, because it provides a memorable framework for reminding developers of how to construct full, usable models.
+
+- this - this keyword refers to an object, that object which is executing the current bit of javascript code. In other words, every javascript function while executing has a reference to its current execution context, called this.
```
Comment (rachaelstanislaw) [inline interview-questions.md:71]: nice research!

**Review by rachaelstanislaw — APPROVED** [review]
excellent work on your first assessment! let me know if you have questions about any of the feedback. then you're good to merge your code to master :)

File: code-challenges.js [hunk 15]
```diff
@@ -10,23 +10,48 @@ var temp1 = 35
...
+  }
+  else if (num == 212) {
+    return "212 is at boiling point";
+  }
+  else if (num > 212) {
+    return (num + " is above boiling point");
+  }
+  else return "error";
+}
+
```
[the remark this answers is not in the document]
Comment (thomasgonda3) [reply code-challenges.js:25]: I just have always done it that way and it seems like less text to type.  I guess I'm leaving out the information on if I want my function to be a const or var when I declare it this way, and I'm not sure what the syntax "function(argument)" will naturally save the function as (const or var).  I notice inside of map and filter you guys still write function this way for some reason sometimes.  I wasn't aware that this arrow syntax was the new way I'll try and do it that way from now on.

**Pull request merged**